File: jan2025/MaxScoreAfterSplittingString.java
import java.util.*;
public class MaxScoreAfterSplittingString {
    public static void main(String[] args) {
        try (Scanner sc = new Scanner(System.in)) {
            String str = sc.nextLine();
            int ones = (int) str.chars().filter(character -> character == '1').count();
            int zeros = 0;
            int i = 0;
            int max = 0;
            while(i<str.length()-1){
                if(str.charAt(i) == '0'){
                    zeros++;
                }else {
                    ones--;
                }
                max = Math.max(max, zeros+ones);
                i++;
            }
            // System.out.println("Max score we can get from string is %d".formatted(max));
            System.out.println(max);
        } catch (Exception e) {
            System.err.println(e.getMessage());
            // TODO: handle exception
        }        
    }
}

File: jan2025/.cph/MaxScoreAfterSplittingString.py
def solution(string:str) -> int:
    ones = string.count('1')
    zeros = 0
    i = 0
    max_score = 0
    while i<len(string)-1:
        if string[i] == '0':
            zeros += 1
        else:    
            ones -= 1
        i += 1
        max_score = max(max_score, zeros+ones)
    return max_score

string = input()
print(solution(string))
        